test/testMinsky.cc

```diff
@@ -1277,3 +1282,613 @@ TEST(TensorOps, evalOpEvaluate)
+    }
+
+    // Test openGroupInCanvas and openModelInCanvas
+    TEST_F(MinskySuite, canvasGroupOperations)
+    {
+      auto g1 = model->addGroup(new Group);
+      g1->addItem(VariablePtr(VariableType::flow, "groupVar"));
+      
+      canvas.item = g1;
+      openGroupInCanvas();
+      EXPECT_TRUE(canvas.model != model);
+      
+      openModelInCanvas();
+      EXPECT_TRUE(canvas.model == model);
+    }
+
+    // Test saveSelectionAsFile
+    TEST_F(MinskySuite, saveSelectionAsFile)
+    {
+      string selFile = "/tmp/test_selection.mky";
+      
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "selVar"));
+      canvas.selection.ensureItemInserted(var1);
+      
+      saveSelectionAsFile(selFile);
+      
+      ifstream f(selFile);
+      EXPECT_TRUE(f.good());
+      f.close();
+      remove(selFile.c_str());
+    }
+
+    // Test saveCanvasItemAsFile
+    TEST_F(MinskySuite, saveCanvasItemAsFile)
+    {
+      string canvasFile = "/tmp/test_canvas_item.mky";
+      
+      auto g1 = model->addGroup(new Group);
+      g1->addItem(VariablePtr(VariableType::flow, "canvasVar"));
+      canvas.item = g1;
+      
+      saveCanvasItemAsFile(canvasFile);
+      
+      ifstream f(canvasFile);
+      EXPECT_TRUE(f.good());
+      f.close();
+      remove(canvasFile.c_str());
+    }
+
+    // Test listAllInstances
+    TEST_F(MinskySuite, listAllInstances)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "instVar"));
+      canvas.item = var1;
+      
+      // Just verify it doesn't crash
+      listAllInstances();
+    }
+
+    // Test initGodleys
+    TEST_F(MinskySuite, initGodleys)
+    {
+      auto g1 = new GodleyIcon;
+      model->addItem(g1);
+      g1->table.resize(3, 3);
+      g1->table.cell(0,1) = "initStock";
+      g1->table.cell(2,1) = "initFlow";
+      g1->update();
+      
+      // Just verify it doesn't crash
+      initGodleys();
+    }
+
+    // Test reloadAllCSVParameters
+    TEST_F(MinskySuite, reloadAllCSVParameters)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::parameter, "csvParam"));
+      
+      // Just verify it doesn't crash
+      reloadAllCSVParameters();
+    }
+
+    // Test redrawAllGodleyTables
+    TEST_F(MinskySuite, redrawAllGodleyTables)
+    {
+      auto g1 = new GodleyIcon;
+      model->addItem(g1);
+      
+      // Just verify it doesn't crash
+      redrawAllGodleyTables();
+    }
+
+    // Test inputWired
+    TEST_F(MinskySuite, inputWired)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "wireVar"));
+      EXPECT_FALSE(inputWired(":wireVar"));
+      
+      auto op1 = model->addItem(OperationPtr(OperationType::time));
+      model->addWire(op1->ports(0), var1->ports(1));
+      EXPECT_TRUE(inputWired(":wireVar"));
+    }
+
+    // Test commandHook
+    TEST_F(MinskySuite, commandHook)
+    {
+      // Test with a generic command
+      bool result = commandHook("minsky.test.command", 0);
```

Comment:
Do pushHistory to intialise the history stack, reset the edit flag, then change the model, eg t+=1, then after the commandHook call, check that the history stack has been pushed, and that the edited flag is set. The return value should be true.